docs/data.rst

```diff
@@ -218,6 +221,30 @@ the following code after the previous call to ``set_dictionary``:
 
 The value of ``my_point`` is now ``{"x":10, "y":200}``
 
+It is possible to obtain the dictoinary of a nested struct using
+`SampleIterator.get_dictionary("member_name")`:
+
+.. testcode::
+
+   for sample in input.valid_data_iterator:
+      point = sample.get_dictionary("my_point")
+
+It is also possible to supply "member_name" as an element of an array (if the
```

Comment:
This looks a bit out of place. We haven't explained yet how to access element of a sequence. I would move this particular case to the next section.

I would also move the the following paragraph ("More generally...") to the previous section ("Using dictionaries vs...").